src/auth/auth.service.ts

```diff
@@ -55,4 +66,43 @@ export class AuthService {
     return null;
   }
   
+  // verify email token
+  async confirmEmail(token: string): Promise<Boolean> {
+    let sanitizedUser = await this.usersService.findOne({emailNonce: token});
+    if (!sanitizedUser) {
+      return false;
+    }
+    return this.usersService.validateEmailNonce(token);
+  }
+
+
+  // request password reset
+  async passwordRequest(passwordRequestDto: PasswordRequestDto): Promise<void> {
+    try {
+      let user = await this.usersService.findOneUnsanitized({email: passwordRequestDto.email})
+      if (user) {
+        let passwordNonce = uuidv4()
+        await this.usersService.updatePasswordNonce(passwordNonce, user.id);
```

Comment:
Generating the password nonce should remain responsibility of the Users service. Please see comment regarding this on the `users.service.ts` file.